pkg/controller/ippool/pool_controller.go

```diff
@@ -0,0 +1,378 @@
+	// and also makes sure we spot when things change that might not trigger a reconciliation.
+	if err = utils.AddPeriodicReconcile(c, utils.PeriodicReconcileTime, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("tigera-ippool-controller failed to create periodic reconcile watch: %w", err)
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &Reconciler{}
+
+type Reconciler struct {
+	config               *rest.Config
+	client               client.Client
+	scheme               *runtime.Scheme
+	watches              map[runtime.Object]struct{}
+	autoDetectedProvider operator.Provider
+	status               status.StatusManager
+}
+
+const (
+	// This label is used to track which IP pools are managed by this controller. Any IP pool
+	// with this label key/value pair is assumed to be solely managed and reconciled by this controller.
+	// We can't use OwnerReferences until this issue is resolved: https://github.com/projectcalico/calico/issues/5715
+	managedByLabel = "app.kubernetes.io/managed-by"
+	managedByValue = "tigera-operator"
+)
+
+// hasOwner returns true if the given IP pool is owned by the given Installation object, and
+// false otherwise.
+func hasOwner(pool *crdv1.IPPool, installation *operator.Installation) bool {
+	// Check for the managed-by label. Once OwnerReferences are supported on projectcalico.org/v3
+	// resources, we can remove this shortcut. Ideally, we would be comparing the OwnerReference to the UID
+	// on the Installation here instead.
+	if val, ok := pool.Labels[managedByLabel]; ok && val == managedByValue {
+		return true
+	}
+	return false
+}
+
+// Reconcile reconciles IP pools in the cluster.
+//
+// - Query desired IP pools (from Installation)
+// - Query existing IP pools owned by this controller
+// - Reconcile the differences
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.V(1).Info("Reconciling IP pools")
+
+	// Get the Installation object - this is the source of truth for IP pools managed by
+	// this controller.
+	installation := &operator.Installation{}
+	if err := r.client.Get(ctx, utils.DefaultInstanceKey, installation); err != nil {
+		if apierrors.IsNotFound(err) {
+			reqLogger.Info("Installation config not found")
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "An error occurred when querying the Installation resource")
+		return reconcile.Result{}, err
+	}
+	r.status.OnCRFound()
+	defer r.status.SetMetaData(&installation.ObjectMeta)
+
+	// This controller relies on the core Installation controller to perform initial defaulting before it can continue.
+	// The core installation controller adds a specific finalizer as part of performing defaulting,
+	// so wait for that before we continue.
+	finalizerExists := false
```

Comment:
I think perhaps we should name this `installationDefaulted` to indicate what we're using this variable for and since we don't really care if the finalizer exists.
If we changed how we determined the Installation resource has been defaulted in the future then the variable wouldn't need to change.
I think this helps make the code self documenting.

Reply:
I considered but didn't name it that way because the installation goes through more than one round of defaulting now, so `installationDefaulted` is ambiguous. We're waiting for the core controller to perform its defaulting, so that this controller can perform additional defaulting.

I think this variable is used only within these several lines, so it's not especially confusing. I could rename it to something like "coreInstallationDefaultingComplete" or "readyToDefaultIPPools" instead, but both of those seem a bit wordy given the limited scope of this variable. 

Maybe a helper function that hides it all away would make this feel better.

Reply:
Yeah I guess this is fine as it is.
I still feel like it would be better to have an ambiguous name rather than one that indicates something we don't really care about (you could say finalizerExists is ambiguous too), but I don't feel strongly about this if you like the way it currently is.